Sources/FuzzILTool/main.swift

```diff
@@ -0,0 +1,268 @@
+    let proto = try Fuzzilli_Protobuf_Program(serializedData: data)
+    var resString = String()
+    dump(proto, to:&resString, maxDepth: 3)
+    return resString
+}
+
+//Convert a serialized protobuf to a FuzzIL program
+public func protobufToProgram(path: String) throws -> Program {
+    let data = try Data(contentsOf: URL(fileURLWithPath: path))
+    let proto = try Fuzzilli_Protobuf_Program(serializedData: data)
+    let program = try Program(from: proto)
+    return program
+}
+
+//Take a program and lifts it to a JS script
+public func progToJSString(prog: Program) -> String{
+    let lifter = JavaScriptLifter(prefix: jsPrefix,
+                                suffix: jsSuffix,
+                                inliningPolicy: NeverInline(),
+                                ecmaVersion: ECMAScriptVersion.es6)
+    let res = lifter.ilSupportLift(prog)
+    return res.trimmingCharacters(in: .whitespacesAndNewlines)
+}
+
+// func pretty print state
+// Really just for dev help, so no need to get super fancy
+public func getPrettyStringProgram(program: Program) -> String {
+    var resString = String()
+    dump(program, to: &resString)
+    return resString.trimmingCharacters(in: .whitespacesAndNewlines)
+}
+
+// Takes all .il.protobuf files in a directory, and lifts them to JS
+// Returns the number of files successfully converted
+public func compileAllProtosToJS(dirPath: String) throws -> Int {
+    let fileEnumerator = FileManager.default.enumerator(atPath: dirPath)
+    var count = 0
+    while let fileName = fileEnumerator?.nextObject() as? String {
+        if !fileName.hasSuffix(protoBufFileExtension) { continue }
+        let fullPath = dirPath + fileName
+        let prog = try protobufToProgram(path: fullPath)
+        let jsProgString = progToJSString(prog: prog)
+        let newFilePath = dirPath + String(fileName.dropLast(protoBufFileExtension.count)) + jsFileExtension
+        try jsProgString.write(to: URL(fileURLWithPath: newFilePath), atomically: false, encoding: String.Encoding.utf8)
+        count += 1
+    }
+    return count
+}
+
+// Provided a directory with a bunch of protobuf files, combine them all into a state file for consumption by Fuzzilli
+public func combineProtobufs(dirPath: String, outputFile: String) throws -> Int {
+    let fileEnumerator = FileManager.default.enumerator(atPath: dirPath)
+    var count = 0
+    var bad_count = 0
+    var buf = Data()
+    while let fileName = fileEnumerator?.nextObject() as? String {
+        if !fileName.hasSuffix(protoBufFileExtension) { continue }
+        let fullPath = dirPath + fileName
+        var prog = Program()
+        do {
+            prog = try protobufToProgram(path: fullPath)
+        } catch{
+            print("Failed to convert to program \(fileName) with error \(error)")
+            bad_count += 1
+            continue
+        }
+        let proto = prog.asProtobuf(with: Optional.none) //Don't use opcache, to keep things simple
```

Comment:
Same as above, but couldn't you also use the `encodeProtobufCorpus` function to avoid code duplication?